Automattic/wp-calypso
# Provision available mailboxes for Professional Email

#### Changes proposed in this Pull Request

This PR adds a new page i.e.  `/email/<domain>/titan/setup-mailbox/<site-slug>` which contains a form for creating a  mailbox as an additional post-signup task for Professional Email opt-ins.

- The form accepts mailbox parameters and sends them via a REST API for provisioning on the backend.
- Adds a parameter to the existing `TitanNewMailboxList` component to support adding a single mailbox
- Adds two react query components, a query that performs mailbox availability lookups, and a mutation that actually creates the mailbox resource if it's available.

This is part of the new effort to add Professional Email as part of the sign-up flow.

*

#### Testing instructions



- Checkout the branch, or use the [live branch](https://calypso.live/?image=registry.a8c.com/calypso/app:build-11347)
- Select a site where you have an existing Professional Email subscription
- Navigate to `/email/<domain>/titan/setup-mailbox/<site-slug>` 
- Confirm that the form is visible
- Confirm that you are redirected to the Email Upsell page when there's no Titan subscription
- Attempt to create a mailbox with an existing name. Confirm that a conflict error is returned
- Attempt to create a mailbox without any free licenses/slots. Confirm that an appropriate error is returned.
- Confirm that a mailbox can be created when all conditions are favourable
- Confirm a redirect to the Email Plan page when successful. NOTE: This is temporary until the Thank You page is ready.

<img width="1065" alt="Screenshot 2021-07-27 at 6 30 58 PM" src="https://user-images.githubusercontent.com/277661/127200620-3ce232e3-d056-4447-9bda-239c998924ba.png">
